### PR TITLE
Release 5.0.0

### PR DESCRIPTION
**Pull request by thiemowmde**
[Bug: T125636](https://phabricator.wikimedia.org/T125636)

### CODE REVIEW

**Comment by JeroenDeDauw** [comment]
Not sure if it's worth [removing the version constant](https://github.com/wmde/WikibaseDataModel/commit/bdc20a2732aeca3fd81d33f0d68de3a360bfd2e6). It is used, for instance here https://github.com/ProjetPP/WikibaseEntityStore/blob/master/src/Cache/EntityIdForTermCache.php#L81

@filbertkm what motivated you in removing it?

**Comment by JeroenDeDauw** [comment]
@thiemowmde I'm a bit confused by a release PR including a release data if you think a release should not be made yet.

**Comment by JeroenDeDauw** [comment]
Having a major release for just those two small things is rather unfortunate. Not this is worth the cost.

**Comment by thiemowmde** [comment]
I hope we can release this today?

I suggest to include all Claim related breaking changes in 5.0.0 (#157, #556, #557), but not remove Entity in the same release. What do you think?

**Comment by Benestar** [comment]
Is there such a hurry that we need to release this today? I'd like to go over the patches that remove legacy code and see if I can get rid of some usages in Wikibase.git. Maybe we can at least wait until after the weekend?

**Comment by thiemowmde** [comment]
We already did that. Claims/getClaims stuff is almost unused, and getting rid of the remaining one or two instances is very easy. Getting rid of Entity is a whole different story. This will not happen within the next 48 hours.

What exactly do you want to work on?

**Comment by filbertkm** [comment]
@JeroenDeDauw we can re-add the version constant, since as you point out, it is still used elsewhere and is useful.

**Comment by thiemowmde** [comment]
@Benestar, @JeroenDeDauw, what's going on here? I carefully arranged the existing pull requests in 5.0.0 and 6.0.0 milestones, hoping we could do a release today. What are all these other commits you two merged **over the weekend**? Where do they come from? Who decided which change should be in which release? Who decided that no other team member with a regular shift should have a chance to look at these changes? Where is the discussion to effectively ditch the milestones and instead do whatever somebody feels like? Who takes responsibility for this? Who will do all the updates that are now required to update all the other components depending on this? Who will review these? In which sprint planning was this discussed?

Pinging @tobijat and @lydiapintscher to let them know we are **again** in the same situation where two people form a group outside of the team and confirm each others patches with nobody else having a chance. I consider this **again** a major blocker for the current sprint.

I lost track and I refuse to continue working on this until people start talking and discussing again.

**Comment by JeroenDeDauw** [comment]
Thiemo, I'm sorry if we upset you in some way.

If I'm not mistaken both me and @Benestar are only intending to help with getting a release going. The things that where done are to me not controversial, and have generally been on the roadmap for a long time. As far as I can tell, they also do not cause any problems in forcing a delay in the release. And for those where I was less confident about it, I did not merge them yet, waiting for a second pair of eyes to look at them.

We never agreed on using the milestones in a formal way. We can release, and have made releases, without merging all items on them first. I'm looking at milestones more as a "partial list of things to look at before release".

I'm more than happy to discuss concrete problems with you, and am pretty sure the same is true for @Benestar. That said, I think it's really not nice towards us to react to our efforts by writing angry comments demanding we explain ourselves and involving other people like this.

**Comment by Benestar** [comment]
Thiemo, I'm sorry but I don't think your overly offending reaction is appropriate in any way. I don't think the way you are communicating in this place is pleasant or nice as Jeroen puts it, it's not nice to react in such an angry way to the work others have done.

**Comment by thiemowmde** [comment]
Blockers for an actual release:
* #607 removed about a dozen public methods. This needs at least much more detailed documentation.
* #615 should be reverted or all issues fixed.
* #609 should be reverted or all issues fixed.
* The https://github.com/wmde/WikibaseDataModel/milestones/5.0.0 milestone and what's in there and what not obviously needs discussion, possibly including a discussion how we deal with milestones in general.

**Comment by thiemowmde** [comment]
So asking questions that would help me to get back on track is offending? What?

I want to add that I find it really, really strange that you are now blaming me for involving other people, when exactly this (not involving the team) is one of the main issues I have.

I really don't know how I should respond to the statement that "milestone" does not mean "milestone" but something else, especially when you are referring to some other "roadmap" in the same comment. What "roadmap"?

**Comment by thiemowmde** [comment]
And it continues:
* While #617 tries to fix some issues, it still has more, but can not be amended because it got merged 36 minutes later with no external review.

File: WikibaseDataModel.php [hunk 1]
```diff
@@ -7,7 +7,10 @@
  * @author Jeroen De Dauw < jeroendedauw@gmail.com >
  */
 
-define( 'WIKIBASE_DATAMODEL_VERSION', '5.0.0 alpha' );
+/**
+ * @deprecated since 5.0
```
Comment (Benestar) [inline WikibaseDataModel.php]: Why is this actually deprecated in here? If I understand Katie's comment correctly we want to keep this in here and not remove it anytime soon.

Comment (thiemowmde) [inline WikibaseDataModel.php]: After #595 this constant is of almost no use any more. It doesn't even tell you if the component is loaded or not. By deprecating it I want to document this fact. This does not mean we will remove it any time soon.

If this is considered a blocker for 5.0 we can remove this line.

File: RELEASE-NOTES.md [hunk 2]
```diff
@@ -17,8 +17,13 @@
     * Added `getIterator` method.
 * Removed `ReferenceList::removeDuplicates`.
 * `ReferenceList::addReference` now throws an `InvalidArgumentException` for negative indices.
-* `Entity` no longer implements `Comparable`
-* Added `EntityDocument::equals`
+* `Entity`, `Item` and `Property` no longer implement `Comparable`.
+    * Added `EntityDocument::equals`.
+    * Changed `equals` methods in `Item` and `Property` to require an `EntityDocument`.
```
Comment (thiemowmde) [inline RELEASE-NOTES.md]: After #621 this needs an update.

File: WikibaseDataModel.php (hunk 1 above)
Comment (JeroenDeDauw) [inline WikibaseDataModel.php]: > It doesn't even tell you if the component is loaded or not.

What? "is loaded"? What does that even mean?

Comment (thiemowmde) [inline WikibaseDataModel.php]: See https://github.com/wmde/WikibaseDataModel/blob/4.4.0/WikibaseDataModel.mw.php#L11 or the discussion in https://gerrit.wikimedia.org/r/268401.

Comment (filbertkm) [inline WikibaseDataModel.php]: @thiemo the constant tells you the version. 

In https://github.com/ProjetPP/WikibaseEntityStore/blob/master/src/Cache/EntityIdForTermCache.php#L81 it is used as part of a cache key, which seems a valid use case to support

File: RELEASE-NOTES.md [hunk 3]
```diff
@@ -16,9 +16,11 @@
     * Added `getIterator` method
 * Removed `ReferenceList::removeDuplicates`
 * `ReferenceList::addReference` now throws an `InvalidArgumentException` for negative indices
-* `EntityDocument` now implements `Comparable`
-    * Added `equals` method
-* Added `copy` to `EntityDocument`
+* Added `EntityDocument::equals`, and `EntityDocument` now implements `Comparable`
```
Comment (Benestar) [inline RELEASE-NOTES.md]: Oxford comma? :D

Comment (thiemowmde) [inline RELEASE-NOTES.md]: Yes, sir.

**Pull request merged**